pipeline/train/train.sh

```diff
@@ -30,42 +35,88 @@ mkdir -p "${model_dir}/tmp"
 
 all_model_metrics=(chrf ce-mean-words bleu-detok)
 
-echo "### Training ${model_dir}"
+echo "### Preparing tsv datasets and config"
+
+# Generate a new OpusTrainer config based on a template to fill paths of the datasets
+new_config="${model_dir}/config.opustrainer.yml"
+cp "configs/opustrainer/${model_type}.yml" "${new_config}"
+
+# Iterate over the training sets
+# split the input string into an array
+IFS=',' read -ra datasets <<< "${train_set_prefixes}"
+IFS=',' read -ra alns <<< "${alignments}"
+# loop through the array and get both value and index
+for index in "${!datasets[@]}"; do
+    train_set_prefix="${datasets[index]}"
+    # OpusTrainer supports only tsv and gzip
+    # TODO: pigz is not found
+    tsv_dataset="${train_set_prefix}.${src}${trg}.tsv" #.gz"
+
+    if [ "${alignments}" != "None" ] ; then
+      train_aln="${alns[index]}"
+      # alignments are currently used only for student that uses one stage of training
+      echo "### Generating tsv dataset with alignments ${alignments}"
+      paste <(${COMPRESSION_CMD} -dc "${train_set_prefix}.${src}.${ARTIFACT_EXT}") \
```

Comment:
Question: I'd be curious how performant this is, and what the memory model of it is. It's not clear to me how much this loads into memory, or how this works from an implementation perspective.

Reply:
This is a common pattern we use throughout the pipeline. I did not dig into implementation details but I didn't have issues with it.